packages/jazz-tools/src/tools/coValues/deepLoading.ts

```diff
@@ -2,11 +2,23 @@ import { SessionID } from "cojson";
 import { ItemsSym } from "../internal.js";
 import { type Account } from "./account.js";
 import { CoFeedEntry } from "./coFeed.js";
-import { type CoKeys, type CoMap } from "./coMap.js";
+import { type CoKeys } from "./coMap.js";
 import { type CoValue, type ID } from "./interfaces.js";
 
 type NotNull<T> = Exclude<T, null>;
 
+/**
+ * Used to check if T is a union type.
+ *
+ * If T is a union type, the left hand side of the extends becomes a union of function types.
+ * The right hand side is always a single function type.
+ */
+type IsUnion<T, U = T> = (T extends any ? (x: T) => void : never) extends (
+  x: U,
+) => void
+  ? false
+  : true;
```

Comment:
Interesting hack!

So if I understand this correctly, both the extends are there in order to force the Typescript compiler to explode every possible type of the union, and the second generic parameter is there to make the compiler infer only one of the possibilities of then union types.

Am I correct?